src/V3Randomize.cpp

```diff
@@ -1331,6 +1331,24 @@ class RandomizeVisitor final : public VNVisitor {
             }
             AstMemberDType* const firstMemberp = unionDtp->membersp();
             return newRandStmtsp(fl, exprp, nullptr, offset, firstMemberp);
+        } else if (AstUnpackArrayDType* const unpackarrayDType = VN_CAST(
+                       memberp ? memberp->subDTypep()->skipRefp() : exprp->dtypep()->skipRefp(),
+                       UnpackArrayDType)) {
+            AstNodeStmt* stmtsp = nullptr;
+            int elemCount = unpackarrayDType->elementsConst();
+            for (int i = 0; i < elemCount; ++i) {
+                AstNodeExpr* elementExprp
+                    = new AstArraySel(fl, stmtsp ? exprp->cloneTree(false) : exprp, i);
```

Comment:
```suggestion
                    = new AstArraySel{fl, stmtsp ? exprp->cloneTree(false) : exprp, i};
```

If this causes a type conversion error, please fix that.